### DISCUSSION TITLE
Can this program pass the VAC?

### DISCUSSION THREAD

**Discussion by WEEHEARTa** [wasiejen/Free-Snap-Tap#11] [Q&A]
As title. This is a very good work, but I really want to know will using software for keyboard input be considered cheating?

**Comment by wasiejen** [comment]
Hello and Welcome.

I think there is no real answer yet to that question.
But I think we can do some Quesstimation: :-)

Let us first answer some basic Questions:

> Can a program or the key events that emulates Snap Tap be detected by any AntiCheat Software (ACS)?
- Defintive yes.
- Especially this program uses the pynput package which is not intented to circumvent ACS. The simulated key events are easily identifiable.

> Can ACS detect Snap Tap which is integrated into the Keyboard Firmware/Software?
- Maybe, but most likely it needs to analyse the Input over longer times, because the changed key events will not be different than the normal key events. This will be connected to uncertainty because maybe someone just has no overlap of key events and might so falsly flagged.

> Is this a form of automation of gameplay?
- Many Games actively try to prevent automation for obvious reasons. But what is automation in games? The most conservative delimitation to automation I came across was that:  "as long as one action in real life is not producing more than one action ingame it is permissable". And I think we are operation in this or very close to that area because every action in real life (key press, key release) results in only one or maybe 2?? (key press of one key releases another and presses the first key - Fast Tapping) action, but which will only result in one action ingame (changed direction).

So, the detection of Snap Tap is relatively easy for programs like this here where there will be send additional key events to realize this functionality. On the other hand it is hard to detect hardware based Snap Tap. If Gaming company now decides to use ACS against Snap Tap it needs to target not only the software based solution but needs to do the hard work and has to detect also the hardware based Snap Tap - which in my opinion needs client side key logging (server side keyboard logging based on timesteps is most likely not enough to recognise this behavior of Snap Tap due to the time windows in which actions that happen close together are simply aggregated together and send together. - pls correct me if I am wrong)

So in my opinion it is very unlikely that effective measures can be taken to prevent the use of Snap Tap completely without just hitting the costumers that do not own the hardware to use this in a harder to detect version. And I do not know if a gaming company wants to differentiate their costumers bases on hardware - might be really bad publicity for them if they do that.

So the ghost might be irrevocably out of the bottle for that. :-)

**Reply by WEEHEARTa** [reply]
Thank you for your thoughtful and detailed answer!

**Comment by wasiejen** [comment]
At the actual moment the answer to the original question is a yes, with new VAC policy in effect and using the recent update V0.8.0. :-)
